torch/_torch_docs.py

```diff
@@ -4625,6 +4624,53 @@ def merge_dicts(*dicts):
     torch.return_types.median(values=tensor([-0.3982,  0.2270,  0.2488,  0.4742]), indices=tensor([1, 4, 4, 3]))
 """.format(**single_dim_common))
 
+add_docstr(torch.nanmedian,
+           r"""
+nanmedian(input) -> Tensor
+
+This is a variant of :func:`torch.median` that "ignores" ``NaN`` values,
+computing the median as if ``NaN`` values in :attr:`input` did not exist.
+If all values are ``NaN`` then the median will be ``NaN``. See the 
+documentation for :func:`torch.median`.
+
+Args:
+    {input}
+
+Example::
+
+    >>> a = torch.tensor([1, float('nan'), 3, 2])
+    >>> a.median()
+    tensor(nan)
+    >>> a.nanmedian()
+    tensor(2.)
+
+.. function:: nanmedian(input, dim=-1, keepdim=False, *, out=None) -> (Tensor, LongTensor)
+
+This is a variant of :func:`torch.median` that "ignores" ``NaN`` values,
+computing the median as if ``NaN`` values in :attr:`input` did not exist.
+If all values in a reduced row are ``NaN`` then the median for that reduction
+will be ``NaN``. See the documentation for :func:`torch.median`.
+
+Args:
+    {input}
+    {dim}
+    {keepdim}
+
+Keyword args:
+    out (tuple, optional): the result tuple of two output tensors (max, max_indices)
```

Comment:
What about something like:

"a tuple of two tensors. The first tensor will be populated with the results of the reduction, and the second tensor will be populated with the original indices of the median elements. The second tensor should be a tensor of long dtype."

This should explain what each tensor holds, and that the second tensor should have the long dtype.

Reply:
Damn, I just noticed how wrong the existing doc was. Why (tuple, optional)? It was copied from torch.max but even for max it is not really optional the indices tensor.

Reply:
Can we have a generic description for this, there are many reduction operations that return a tuple of the reduced values and its indices?

Reply:
Sure. Why not?